src/test/java/net/minecraftforge/test/CustomEntitySelectorTest.java

```diff
@@ -0,0 +1,69 @@
+package net.minecraftforge.test;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Lists;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraftforge.fml.common.IEntitySelectorFactory;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+@Mod(modid = "customentityselectortest", name = "Custom Entity Selector Test", version = "1.0", acceptableRemoteVersions = "*")
+public class CustomEntitySelectorTest
+{
+
+    @Mod.EventHandler
+    public void init(FMLInitializationEvent event)
+    {
+        GameRegistry.registerEntitySelector(new EntitySelectorFactory(), "forge:min_health");
+    }
+
+    private class EntitySelectorFactory implements IEntitySelectorFactory
+    {
+
+        @Nonnull @Override
+        public List<Predicate<Entity>> createPredicates(Map<String, String> arguments, String mainSelector, ICommandSender sender, Vec3d position)
+        {
+            List<Predicate<Entity>> predicates = Lists.newArrayList();
+            String health = arguments.get("forge:min_health");
+            if (health != null)
```

Comment:
Vanilla sometimes uses my approach, sometimes yours. I decided to go with this as I personally think it's easier to understand. 
But as modders might take this as an example and as every (tiny) performance increase counts, I changed it according to your suggestions now :)